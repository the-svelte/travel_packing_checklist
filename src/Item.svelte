<script>
  import {blurOnKey} from './util';

  // { id: string, name: string, packed: boolean }
  export let item;

  let editing = false;
</script>

<li class="list-group-item">
  <div class="d-flex align-items-center justify-content-between">
    <div class="">
      <input
        class="form-check-input"
        type="checkbox"
        id=""
        bind:checked={item.packed}
      >
    </div>
    <div class="flex-grow-1 px-4">
      {#if editing}
        <input
          type="text"
          class="form-control form-control-sm"
          autofocus
          bind:value={item.name}
          on:blur={() => editing = false}
          on:keydown={blurOnKey}
        />
      {:else}
        <label
          class="{item.packed ? 'text-decoration-line-through' : ''}"
          for=""
          on:click={() => editing = true}
        >{item.name}</label>
      {/if}
    </div>
    <div class="">
      <button
        type="button"
        class="btn btn-outline-secondary btn-sm"
      >Delete</button>
    </div>
  </div>
</li>
