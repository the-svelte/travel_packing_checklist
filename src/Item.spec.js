import {cleanup, render} from '@testing-library/svelte';

import Item from './Item.svelte';

describe('Item', () => {
  const categoryId = 1;
  const dnd = {};
  const item = {id: 2, name: 'socks', packed: false};

  afterEach(cleanup);

  test('should render', () => {
    const {getByTestId, getByText} = render(Item, {categoryId, dnd, item});
    const checkbox = document.querySelector('input[type="checkbox"]');

    expect(checkbox).not.toBeNull();
    expect(getByText(item.name));
    expect(getByTestId('delete'));
  });

  test('should match snapshot', () => {
    const {container} = render(Item, {categoryId, dnd, item});

    expect(container).toMatchSnapshot();
  })
});
