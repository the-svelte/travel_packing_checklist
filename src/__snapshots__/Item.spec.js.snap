// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<body>
  <div>
    <li
      class="list-group-item"
    >
      <div
        class="d-flex align-items-center justify-content-between"
      >
        <div
          class=""
        >
          <input
            class="form-check-input"
            id=""
            type="checkbox"
          />
        </div>
         
        <div
          class="flex-grow-1 px-4"
        >
          <label
            class=""
            for=""
          >
            socks
          </label>
        </div>
         
        <div
          class=""
        >
          <button
            class="btn btn-sm text-muted"
            data-testid="delete"
            type="button"
          >
            <svg
              class="bi bi-x-square"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
              />
              <path
                d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
              />
            </svg>
          </button>
        </div>
      </div>
    </li>
  </div>
</body>
`;
